pkg/tsdb/azuremonitor/azure-log-analytics-datasource.go

```diff
@@ -0,0 +1,353 @@
+		if resultFormat == "" {
+			resultFormat = "time_series"
+		}
+
+		urlComponents := map[string]string{}
+		urlComponents["subscription"] = fmt.Sprintf("%v", query.Model.Get("subscription").MustString())
+		urlComponents["workspace"] = fmt.Sprintf("%v", azureLogAnalyticsTarget["workspace"])
+		apiURL := fmt.Sprintf("%s/query", urlComponents["workspace"])
+
+		params := url.Values{}
+		rawQuery, err := KqlInterpolate(query, timeRange, fmt.Sprintf("%v", azureLogAnalyticsTarget["query"]), "TimeGenerated")
+		if err != nil {
+			return nil, err
+		}
+		params.Add("query", rawQuery)
+
+		azureLogAnalyticsQueries = append(azureLogAnalyticsQueries, &AzureLogAnalyticsQuery{
+			RefID:        query.RefId,
+			ResultFormat: resultFormat,
+			URL:          apiURL,
+			Params:       params,
+			Target:       params.Encode(),
+		})
+	}
+
+	return azureLogAnalyticsQueries, nil
+}
+
+func (e *AzureLogAnalyticsDatasource) executeQuery(ctx context.Context, query *AzureLogAnalyticsQuery, queries []*tsdb.Query, timeRange *tsdb.TimeRange) (*tsdb.QueryResult, error) {
+	queryResult := &tsdb.QueryResult{Meta: simplejson.New(), RefId: query.RefID}
+
+	req, err := e.createRequest(ctx, e.dsInfo)
+	if err != nil {
+		queryResult.Error = err
+		return queryResult, nil
+	}
+
+	req.URL.Path = path.Join(req.URL.Path, query.URL)
+	req.URL.RawQuery = query.Params.Encode()
+
+	span, ctx := opentracing.StartSpanFromContext(ctx, "azure log analytics query")
+	span.SetTag("target", query.Target)
+	span.SetTag("from", timeRange.From)
+	span.SetTag("until", timeRange.To)
+	span.SetTag("datasource_id", e.dsInfo.Id)
+	span.SetTag("org_id", e.dsInfo.OrgId)
+
+	defer span.Finish()
+
+	if err := opentracing.GlobalTracer().Inject(
+		span.Context(),
+		opentracing.HTTPHeaders,
+		opentracing.HTTPHeadersCarrier(req.Header)); err != nil {
+		queryResult.Error = err
+		return queryResult, nil
+	}
+
+	azlog.Debug("AzureLogAnalytics", "Request ApiURL", req.URL.String())
+	res, err := ctxhttp.Do(ctx, e.httpClient, req)
+	if err != nil {
+		queryResult.Error = err
+		return queryResult, nil
+	}
+
+	data, err := e.unmarshalResponse(res)
+	if err != nil {
+		queryResult.Error = err
+		return queryResult, nil
+	}
+
+	azlog.Debug("AzureLogsAnalytics", "Response", queryResult)
+
+	if query.ResultFormat == "table" {
+		queryResult.Tables, queryResult.Meta, err = e.parseToTables(data, query.Params.Get("query"))
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		queryResult.Series, queryResult.Meta, err = e.parseToTimeSeries(data, query.Params.Get("query"))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return queryResult, nil
+}
+
+func (e *AzureLogAnalyticsDatasource) createRequest(ctx context.Context, dsInfo *models.DataSource) (*http.Request, error) {
+	u, _ := url.Parse(dsInfo.Url)
+	u.Path = path.Join(u.Path, "render")
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		azlog.Error("Failed to create request", "error", err)
```

Comment:
```suggestion
		azlog.Debug("Failed to create request", "error", err)
```
Logging error and returning an error is usually not a good idea since we dont know how the caller wants to handle the error. I suggest you move this to debug level if you want to keep it.

Reply:
Then I should do the same for the `unmarshalResponse` function too? There are two error logging calls there where an error is returned.

Reply:
Sounds good. I saw a few more bug I thought mentioning it twice would be enough. yey/ney?